code/modules/urist/modules/shipbattles/overmapships/components/advanced_components.dm

```diff
@@ -32,30 +32,30 @@
 	boarding_failure_chance = 0 //Zero this var out.
 
 	if(!bypass_shields)
-		for(var/obj/machinery/power/shield_generator/S in SSmachines.machinery) //Calculate our failure chance.
+		for(var/obj/machinery/power/shield_generator/S in SSmachines.machinery) //Calculate our failure chance. //TODO: Bind shield gens per combat ship
 			if(S.z in GLOB.using_map.station_levels)
 				if(S.running == SHIELD_RUNNING)
 					boarding_failure_chance += 25 // four shield generators to TOTALLY block boarding.
```

Comment:
```suggestion
		for(var/obj/machinery/power/shield_generator/S in SSmachines.machinery) //Calculate our failure chance.
			if(S.z in mastership.target_ship.map_z)
				if(S.running == SHIELD_RUNNING)
					boarding_failure_chance += 25 // four shield generators to TOTALLY block boarding.
```
Can be applied to all the other TODOs

Reply:
Ah! I was not aware this var existed! This helps a lot later down the road! Thanks a lot
Removed all TODO's and temporary `using_map` constraints in d8e9c2aa30c7b9268b2b1db6998bf43ce1472aa0